runtime/src/bank.rs

```diff
@@ -4142,35 +4128,35 @@ impl Bank {
             .iter()
             .zip(execution_results)
             .map(|(tx, execution_result)| {
-                let (execution_status, durable_nonce_fee) = match &execution_result {
-                    TransactionExecutionResult::Executed { details, .. } => {
-                        Ok((&details.status, details.durable_nonce_fee.as_ref()))
-                    }
-                    TransactionExecutionResult::NotExecuted(err) => Err(err.clone()),
-                }?;
-                let is_nonce = durable_nonce_fee.is_some();
-
                 let message = tx.message();
-                let fee_details = self.fee_structure.calculate_fee_details(
-                    message,
-                    &process_compute_budget_instructions(message.program_instructions_iter())
-                        .unwrap_or_default()
-                        .into(),
-                    self.feature_set
-                        .is_active(&include_loaded_accounts_data_size_in_fee_calculation::id()),
-                );
-
-                self.check_execution_status_and_charge_fee(
-                    message,
-                    execution_status,
-                    is_nonce,
-                    fee_details.total_fee(
+                let (execution_status, is_nonce, lamports_per_signature) = self
+                    .get_details_from_execution_result(
+                        execution_result,
+                        message.recent_blockhash(),
+                    )?;
+
+                if !FeeStructure::to_clear_transaction_fee(lamports_per_signature) {
```

Comment:
I realize it'd be changing the logic. but are there any cases where this will be non-zero in one case and zero in another?
i.e. could we have an outer if and skipp the entire function  body if the bank's lamports per signature is 0?

Reply:
Nonce transactions might? theoretically at least. 

I also feel keeping this 'set transaction fee to zero if lamports_per_signature is zero' hack closer to "calculate_fe()" is less confusing.